### PR TITLE
bump project version to vaadin 24

### PR DESCRIPTION
**Pull request by ZheSun88**

### CODE REVIEW

**Comment by ZheSun88** [comment]
currently, local build failed on 
```
 [36mvite v4.1.1 building for production...
[ERROR] transforming...
[ERROR] ✓ 9 modules transformed.
[ERROR] Could not resolve "./vaadin-featureflags.ts" from "frontend/generated/vaadin.ts"
[ERROR] file: C:/WorkFolder/docs/frontend/generated/vaadin.ts
[ERROR] error during build:
[ERROR] RollupError: Could not resolve "./vaadin-featureflags.ts" from "frontend/generated/vaadin.ts"
[ERROR]     at error (file:///C:/WorkFolder/docs/node_modules/vite/node_modules/rollup/dist/es/shared/rollup.js:2091:30)
[ERROR]     at ModuleLoader.handleInvalidResolvedId (file:///C:/WorkFolder/docs/node_modules/vite/node_modules/rollup/dist/es/shared/rollup.js:23325:24)
[ERROR]     at file:///C:/WorkFolder/docs/node_modules/vite/node_modules/rollup/dist/es/shared/rollup.js:23287:26
[ERROR] frontend/demo/component/grid/grid-item-details.ts(45,38): error TS2322: Type 'Person | null | undefined' is not assignable to type 'Person'.
[ERROR]   Type 'undefined' is not assignable to type 'Person'.
[ERROR] frontend/demo/foundation/icons-preview.ts(23,17): error TS2339: Property 'applyIcon' does not exist on type 'VaadinIconset'.
[ERROR] frontend/demo/foundation/icons-preview.ts(24,19): error TS2339: Property 'applyIcon' does not exist on type 'VaadinIconset'.
```

**Comment by web-padawan** [comment]
@ZheSun88 The following components issues are now addressed:

```
[ERROR] frontend/demo/component/grid/grid-item-details.ts(45,38): error TS2322: Type 'Person | null | undefined' is not assignable to type 'Person'.
[ERROR]   Type 'undefined' is not assignable to type 'Person'.
[ERROR] frontend/demo/foundation/icons-preview.ts(23,17): error TS2339: Property 'applyIcon' does not exist on type 'VaadinIconset'.
[ERROR] frontend/demo/foundation/icons-preview.ts(24,19): error TS2339: Property 'applyIcon' does not exist on type 'VaadinIconset'.
```

File: src/main/java/com/vaadin/demo/component/upload/UploadExamplesI18N.java [hunk 1]
```diff
@@ -19,7 +19,6 @@ public UploadExamplesI18N() {
                 .setMany("Drop files here"));
         setAddFiles(new AddFiles().setOne("Upload File...")
                 .setMany("Upload Files..."));
-        setCancel("Cancel");
```
Comment (mshabarov) [inline src/main/java/com/vaadin/demo/component/upload/UploadExamplesI18N.java:22]: Why is this removed?

**Review by mshabarov — APPROVED** [review]

**Comment by mshabarov** [comment]
> currently, local build failed on
> 
> ```
>  [36mvite v4.1.1 building for production...
> [ERROR] transforming...
> [ERROR] ✓ 9 modules transformed.
> [ERROR] Could not resolve "./vaadin-featureflags.ts" from "frontend/generated/vaadin.ts"
> [ERROR] file: C:/WorkFolder/docs/frontend/generated/vaadin.ts
> [ERROR] error during build:
> [ERROR] RollupError: Could not resolve "./vaadin-featureflags.ts" from "frontend/generated/vaadin.ts"
> [ERROR]     at error (file:///C:/WorkFolder/docs/node_modules/vite/node_modules/rollup/dist/es/shared/rollup.js:2091:30)
> [ERROR]     at ModuleLoader.handleInvalidResolvedId (file:///C:/WorkFolder/docs/node_modules/vite/node_modules/rollup/dist/es/shared/rollup.js:23325:24)
> [ERROR]     at file:///C:/WorkFolder/docs/node_modules/vite/node_modules/rollup/dist/es/shared/rollup.js:23287:26
> [ERROR] frontend/demo/component/grid/grid-item-details.ts(45,38): error TS2322: Type 'Person | null | undefined' is not assignable to type 'Person'.
> [ERROR]   Type 'undefined' is not assignable to type 'Person'.
> [ERROR] frontend/demo/foundation/icons-preview.ts(23,17): error TS2339: Property 'applyIcon' does not exist on type 'VaadinIconset'.
> [ERROR] frontend/demo/foundation/icons-preview.ts(24,19): error TS2339: Property 'applyIcon' does not exist on type 'VaadinIconset'.
> ```

Should be fixed by https://github.com/vaadin/hilla/pull/743
